WordPress/Classes/ViewRelated/Reader/DetailWebView/ReaderDetailToolbar.swift

```diff
@@ -0,0 +1,323 @@
+import UIKit
+
+class ReaderDetailToolbar: UIView, NibLoadable {
+    @IBOutlet weak var dividerView: UIView!
+    @IBOutlet weak var saveForLaterButton: UIButton!
+    @IBOutlet weak var reblogButton: PostMetaButton!
+    @IBOutlet weak var commentButton: PostMetaButton!
+    @IBOutlet weak var likeButton: PostMetaButton!
+
+    /// The reader post that the toolbar interacts with
+    private var post: ReaderPost?
+
+    /// The VC where the toolbar is inserted
+    private weak var viewController: UIViewController?
+
+    /// An observer of the number of likes of the post
+    private var likeCountObserver: NSKeyValueObservation?
+
+    /// An observer of the number of likes of the post
+    private var commentCountObserver: NSKeyValueObservation?
+
+    /// If we should hide the comments button
+    var shouldHideComments = false
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+
+        applyStyles()
+
+        adjustInsetsForTextDirection()
+
+        prepareActionButtonsForVoiceOver()
+    }
+
+    func configure(for post: ReaderPost, in viewController: UIViewController) {
+        self.post = post
+        self.viewController = viewController
+
+        likeCountObserver = post.observe(\.likeCount, options: .new) { [weak self] _, _ in
+            self?.configureLikeActionButton(true)
+        }
+
+        commentCountObserver = post.observe(\.commentCount, options: .new) { [weak self] _, _ in
+            self?.configureCommentActionButton()
+        }
+
+        configureActionButtons()
+    }
+
+    deinit {
+        likeCountObserver?.invalidate()
+        commentCountObserver?.invalidate()
+    }
+
+    // MARK: - Actions
+
+    @IBAction func didTapSaveForLater(_ sender: Any) {
+        guard let readerPost = post, let context = readerPost.managedObjectContext,
+            let viewController = viewController as? UIViewController & UIViewControllerTransitioningDelegate else {
+            return
+        }
+
+        if !readerPost.isSavedForLater {
+            FancyAlertViewController.presentReaderSavedPostsAlertControllerIfNecessary(from: viewController)
+        }
+
+        ReaderSaveForLaterAction().execute(with: readerPost, context: context, origin: .postDetail) { [weak self] in
+            self?.saveForLaterButton.isSelected = readerPost.isSavedForLater
+            self?.prepareActionButtonsForVoiceOver()
+        }
+    }
+
+    @IBAction func didTapReblog(_ sender: Any) {
+        guard let post = post, let viewController = viewController else {
+            return
+        }
+
+        ReaderReblogAction().execute(readerPost: post, origin: viewController, reblogSource: .detail)
+    }
+
+    @IBAction func didTapComment(_ sender: Any) {
+        guard let post = self.post, let viewController = viewController else {
```

Comment:
can remove `self.` here

Reply:
Done in 3249f7eded6c8ac62b92f76b4223d5f268b9c0d9